src/controllers/AuthenticatingController.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace sphereon\craft\controllers;
+
+use Craft;
+use craft\web\Controller;
+
+abstract class AuthenticatingController extends Controller
+{
+    protected function setAuthorization(?\craft\base\Model $settings, \craft\web\Request|\craft\console\Request|\yii\web\Request|\yii\console\Request $request)
+    {
+        $accessToken = $settings->getStaticAccessToken();
+        if ($accessToken == null || strlen($accessToken) == 0) {
+            $msg = 'A static access token must be configured in the plugin configuration';
```

Comment:
It should still be possible to not have an access token. This depends on whether the agent requires authentication or not. Suggest to only set the bearer in case the access token is set and has a length > 0